backend/src/auth/auth.controller.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  Controller,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  Inject,
+  Request,
+  Get,
+  HttpStatus,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { AuthService } from './auth.service';
+import { RequestWithRoles, RolesRequest } from './interface';
+import { AppLogger } from '../logger/logger.service';
+
+@Controller('auth')
+@ApiTags('Auth API')
+@UseInterceptors(ClassSerializerInterceptor)
+export class AuthController {
+  constructor(
+    @Inject(AuthService) private readonly authService: AuthService,
+    private readonly logger: AppLogger,
+  ) {
+    this.logger.setContext(AuthController.name);
+  }
+
+  @ApiOperation({
+    summary: 'Verify the role and sent user data',
+    description: 'Returns the user data with the request',
+  })
+  @ApiResponse({
+    status: HttpStatus.OK,
+  })
+  @Get()
+  async checkUserPermissions(
+    @Request() req: RequestWithRoles,
+  ): Promise<RolesRequest> {
+    this.logger.log(
+      `${req.method}: ${req.url} - ${req.role} - ${req.username}`,
+    );
+    const member = await this.authService.getLoggedInUser(req);
+    const supervisor = await this.authService.verifySupervisor(req);
+    return {
+      role: req?.role,
+      program: req?.program,
+      username: `${req?.username}`,
+      idir: req?.idir,
+      member: member !== null,
+      supervisor: supervisor,
```

Comment:
Might make sense to use an array for Roles but will require more extensive changes. Using role for setting permissions elsewhere, but on this endpoint I am returning member/supervisor boolean in order to know which routes should be available in the user navigation menu.